### PR TITLE
Update deps. Switch to storybook 8 and pnpm 9

### PR DESCRIPTION
**Pull request by rustworthy**
This suggests to bump majors of pnpm and storybook. 
The latter implies updating some package and switch to new ones (e.g. to storybook-addon-remix-react-router and @testing-library/react).  Some changes were done by the storybook migration wizard, but some had to be manual. We are also updating cjs modules to ts as warnings in the console output would advise.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,5 +91,5 @@
   "msw": {
     "workerDirectory": "public"
   },
-  "packageManager": "pnpm@8.14.1"
```
Comment (bartstc) [inline package.json:94]: Please update it also in `.devcontainer/Dockerfile` so that in the dev container we might use the same PNPM version.

**Review by bartstc — COMMENTED** [review]
Looks good, just one small suggestion ;)

**Comment by bartstc** [comment]
I would be grateful if similar changes were included in the `basic` version/branch in a separate PR. As you already did that for the `core` version, you are pretty up to date with required changes and can manage to do so with ease :).

Of course, I won't block this PR if you don't do that. It will still consume some of your time after all. In that case I will do this on my own in the (hope near) future.

**Comment by rustworthy** [comment]
> I would be grateful if similar changes were included in the `basic` version/branch in a separate PR. As you already did that for the `core` version, you are pretty up to date with required changes and can manage to do so with ease :).
> 
> Of course, I won't block this PR if you don't do that. It will still consume some of your time after all. In that case I will do this on my own in the (hope near) future.

no  problem, I will create a chaser

**Pull request merged**